docs/source/user_guide.rst

```diff
@@ -0,0 +1,727 @@
+                random.random()
+                for _ in range(self.size)
+            ]
+
+    embedder = RandomEmbedder(10)
+    vector = embedder.embed_query("some text")
+
+
+Other Vector Retriever Configuration
+----------------------------------------
+
+Often, not all node properties are pertinent for the RAG context; only a selected few are relevant
+for inclusion in the LLM prompt context. You can specify which properties to return
+using the `return_properties` parameter:
+
+.. code:: python
+
+    from neo4j_genai.retrievers import VectorRetriever
+
+    retriever = VectorRetriever(
+        driver,
+        index_name=POSTER_INDEX_NAME,
+        embedder=embedder,
+        return_properties=["title"],
+    )
+
+
+Pre-Filters
+-----------------------------
+
+When performing a similarity search, one may have constraints to apply.
+For instance, filtering out movies released before 2000. This can be achieved
+using `filters`.
+
+.. note::
+
+    Filters are implemented for all retrievers except the Hybrid retrievers.
+    The documentation below is not valid for external retrievers, which use
+    their own filter syntax (see :ref:`vector-databases-section`).
+
+
+.. code:: python
+
+    from neo4j_genai.retrievers import VectorRetriever
+
+    retriever = VectorRetriever(
+        driver,
+        index_name=POSTER_INDEX_NAME,
+    )
+
+    filters = {
+        "year": {
+            "$gte": 2000,
+        }
+    }
+
+    query_text = "How do I do similarity search in Neo4j?"
+    retriever_result = retriever.search(query_text=query_text, filters=filters)
+
+.. warning::
+
+    When using filters, the similarity search bypasses the vector index and instead utilizes
+    an exact match algorithm
+    Ensure that the pre-filtering is stringent enough to prevent query overload.
+
+The currently supported operators are:
+
+- `$eq`: equal.
+- `$ne`: not equal.
+- `$lt`: less than.
+- `$lte`: less than or equal to.
+- `$gt`: greater than.
+- `$gte`: greater than or equal to.
+- `$between`: between.
+- `$in`: value is in a given list.
+- `$nin`: not in.
+- `$like`: LIKE operator case-sensitive.
+- `$ilike`: LIKE operator case-insensitive.
+
+
+Here are examples of valid filter syntaxes and their meaning:
+
+.. list-table:: Filters syntax
+   :widths: 80 80
+   :header-rows: 1
+
+   * - Filter
+     - Meaning
+   * - {"year": 1999}
+     - year = 1999
+   * - {"year": {"$eq": 1999}}
+     - year = 1999
+   * - {"year": 2000, "title": "The Matrix"}
+     - year = 1999 AND title = "The Matrix"
+   * - {"$and": [{"year": 2000}, {"title": "The Matrix"}]}
+     - year = 1999 AND title = "The Matrix"
+   * - {"$or": [{"title": "The Matrix Revolution"}, {"title": "The Matrix"}]}
+     - title = "The Matrix" OR title = "The Matrix Revolution"
+   * - {"title": {"$like": "The Matrix"}}
+     - title CONTAINS "The Matrix"
+   * - {"title": {"$ilike": "the matrix"}}
+     - toLower(title) CONTAINS "The Matrix"
+
+
+See also:ref:`vectorretriever`.
```

Comment:
There's an issue with this hyperlink

Reply:
I thought I had checked all of them... Thanks for double checking, indeed adding the space is enough to fix it.